--- amarok-1.4.8/amarok/src/devicemanager.cpp	2007-12-17 17:53:19.000000000 +0200
+++ amarok-1.4.8/amarok/src/devicemanager.cpp	2008-02-26 20:48:26.000000000 +0200
@@ -65,7 +65,7 @@
             QDataStream arg(data, IO_WriteOnly);
             QStringList result;
             arg << 5;
-            if (!m_dc->call("kded", "mediamanager", "fullList()", data, replyType, replyData, false, 5000))
+            if (!m_dc->call("kded", "mediamanager", "fullList()", data, replyType, replyData, false, -1))
             {
                 debug() << "During DeviceManager init, error during DCOP call" << endl;
             }
