diff -uNr amarok-1.4.8_src/amarok/src/device/massstorage/Makefile.am amarok-1.4.8_dst/amarok/src/device/massstorage/Makefile.am
--- amarok-1.4.8_src/amarok/src/device/massstorage/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/device/massstorage/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -14,7 +14,7 @@
 
 libamarok_massstorage_device_la_LIBADD = \
     $(top_builddir)/amarok/src/libamarok.la \
-    -lkdecore
+    $(LIB_KDECORE) $(LIB_QT)
 
 libamarok_massstorage_device_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/device/nfs/Makefile.am amarok-1.4.8_dst/amarok/src/device/nfs/Makefile.am
--- amarok-1.4.8_src/amarok/src/device/nfs/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/device/nfs/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -14,7 +14,7 @@
 
 libamarok_nfs_device_la_LIBADD = \
     $(top_builddir)/amarok/src/libamarok.la \
-    -lkdecore
+    $(LIB_KDECORE) $(LIB_QT)
 
 libamarok_nfs_device_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/device/smb/Makefile.am amarok-1.4.8_dst/amarok/src/device/smb/Makefile.am
--- amarok-1.4.8_src/amarok/src/device/smb/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/device/smb/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -14,7 +14,7 @@
 
 libamarok_smb_device_la_LIBADD = \
     $(top_builddir)/amarok/src/libamarok.la \
-    -lkdecore
+    $(LIB_KDECORE) $(LIB_QT)
 
 libamarok_smb_device_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/engine/yauap/Makefile.am amarok-1.4.8_dst/amarok/src/engine/yauap/Makefile.am
--- amarok-1.4.8_src/amarok/src/engine/yauap/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/engine/yauap/Makefile.am	2008-02-27 03:52:58.000000000 +0200
@@ -13,7 +13,7 @@
     $(top_builddir)/amarok/src/libamarok.la \
     $(top_builddir)/amarok/src/plugin/libplugin.la \
     $(LIB_YAUAP)\
-    $(LIB_KDECORE)
+    $(LIB_KDECORE) $(LIB_QT)
 
 
 libamarok_yauap_engine_plugin_la_SOURCES = \
diff -uNr amarok-1.4.8_src/amarok/src/engine/void/Makefile.am amarok-1.4.8_dst/amarok/src/engine/void/Makefile.am
--- amarok-1.4.8_src/amarok/src/engine/void/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/engine/void/Makefile.am	2008-02-27 03:51:55.000000000 +0200
@@ -9,7 +9,7 @@
 libamarok_void_engine_plugin_la_LIBADD = \
     $(top_builddir)/amarok/src/libamarok.la \
     $(top_builddir)/amarok/src/plugin/libplugin.la \
-    $(LIB_KDECORE)
+    $(LIB_KDECORE) $(LIB_QT)
 
 libamarok_void_engine_plugin_la_SOURCES = \
     void-engine.cpp
diff -uNr amarok-1.4.8_src/amarok/src/engine/xine/Makefile.am amarok-1.4.8_dst/amarok/src/engine/xine/Makefile.am
--- amarok-1.4.8_src/amarok/src/engine/xine/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/engine/xine/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -12,7 +12,7 @@
     $(top_builddir)/amarok/src/libamarok.la \
     $(top_builddir)/amarok/src/plugin/libplugin.la \
     -lkdeui -lkdecore \
-    $(XINE_LIBS)
+    $(XINE_LIBS) $(LIB_QT)
 
 libamarok_xine_engine_la_SOURCES = \
     xine-scope.c \
diff -uNr amarok-1.4.8_src/amarok/src/konquisidebar/Makefile.am amarok-1.4.8_dst/amarok/src/konquisidebar/Makefile.am
--- amarok-1.4.8_src/amarok/src/konquisidebar/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/konquisidebar/Makefile.am	2008-02-27 04:00:19.000000000 +0200
@@ -5,7 +5,7 @@
 
 konqsidebar_universalamarok_la_SOURCES = universalamarok.cpp amarokdcopiface.stub
 konqsidebar_universalamarok_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries) -lkonqsidebarplugin
-konqsidebar_universalamarok_la_LIBADD = $(LIB_KHTML) $(LIB_KPARTS) $(LIB_KFILE)
+konqsidebar_universalamarok_la_LIBADD = $(LIB_KHTML) $(LIB_QT) $(LIB_KDECORE) -lDCOP $(LIB_KDEUI)
 
 
 amarokdcopiface_DIR=$(top_srcdir)/amarok/src/amarokcore
diff -uNr amarok-1.4.8_src/amarok/src/Makefile.am amarok-1.4.8_dst/amarok/src/Makefile.am
--- amarok-1.4.8_src/amarok/src/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/Makefile.am	2008-02-27 03:59:57.000000000 +0200
@@ -158,7 +158,7 @@
     $(top_builddir)/amarok/src/statusbar/libstatusbar.la \
     $(top_builddir)/amarok/src/metadata/libmetadata.la \
     $(top_builddir)/amarok/src/magnatunebrowser/libmagnatunebrowser.la \
-    $(LIB_KUTILS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KHTML) $(LIB_KNEWSTUFF) \
+    $(LIB_QT) $(LIB_KPARTS) -lDCOP -lkdefx $(KDE_MT_LIBS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KHTML) $(LIB_KNEWSTUFF) \
     $(TAGLIB_LIBS) $(gl_libs) $(LIB_SQLITE) $(LIB_SQLITE_LOCAL) \
     $(LIB_TUNEPIMP) \
     $(mysql_libs) \
@@ -196,11 +196,8 @@
     $(top_builddir)/amarok/src/plugin/libplugin.la \
     $(top_builddir)/amarok/src/statusbar/libstatusbar.la \
     $(top_builddir)/amarok/src/metadata/libmetadata.la \
-    $(LIB_KUTILS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KHTML) $(LIB_KNEWSTUFF) \
-    $(TAGLIB_LIBS) $(gl_libs) $(LIB_SQLITE) $(LIB_SQLITE_LOCAL) $(EXSCALIBAR_LIBS) \
-    $(LIB_TUNEPIMP) \
-    $(mysql_libs) \
-    $(postgresql_libs)
+    $(LIB_KDECORE) 
+    $(EXSCALIBAR_LIBS)
 
 amarokapp_LDFLAGS = \
     $(all_libraries) \
diff -uNr amarok-1.4.8_src/amarok/src/mediadevice/daap/Makefile.am amarok-1.4.8_dst/amarok/src/mediadevice/daap/Makefile.am
--- amarok-1.4.8_src/amarok/src/mediadevice/daap/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/mediadevice/daap/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -17,7 +17,7 @@
 libamarok_daap_mediadevice_la_LIBADD  = \
     $(top_builddir)/amarok/src/libamarok.la \
     $(top_builddir)/amarok/src/mediadevice/daap/daapreader/libdaapreader.la \
-    $(DNSSD_LIBS) ${LIB_KDEUI} ${LIB_KDECORE} ${LIB_KIO}
+    $(DNSSD_LIBS) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KIO) $(LIB_QT)
 
 libamarok_daap_mediadevice_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/mediadevice/generic/Makefile.am amarok-1.4.8_dst/amarok/src/mediadevice/generic/Makefile.am
--- amarok-1.4.8_src/amarok/src/mediadevice/generic/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/mediadevice/generic/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -15,7 +15,7 @@
 
 libamarok_generic_mediadevice_la_LIBADD  = \
     $(top_builddir)/amarok/src/libamarok.la \
-    -lkdeui -lkdecore -lkio
+    $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KIO) $(LIB_QT)
 
 libamarok_generic_mediadevice_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/mediadevice/ifp/Makefile.am amarok-1.4.8_dst/amarok/src/mediadevice/ifp/Makefile.am
--- amarok-1.4.8_src/amarok/src/mediadevice/ifp/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/mediadevice/ifp/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -15,7 +15,7 @@
 libamarok_ifp_mediadevice_la_LIBADD = \
     $(top_builddir)/amarok/src/libamarok.la \
     -lusb -lifp \
-    -lkdeui -lkdecore -lkio
+    $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KIO) $(LIB_QT)
 
 libamarok_ifp_mediadevice_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/mediadevice/ipod/Makefile.am amarok-1.4.8_dst/amarok/src/mediadevice/ipod/Makefile.am
--- amarok-1.4.8_src/amarok/src/mediadevice/ipod/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/mediadevice/ipod/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -16,7 +16,7 @@
 METASOURCES = AUTO
 
 libamarok_ipod_mediadevice_la_LIBADD  = \
-    $(top_builddir)/amarok/src/libamarok.la
+    $(top_builddir)/amarok/src/libamarok.la $(LIB_QT) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KIO)
 
 libamarok_ipod_mediadevice_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/mediadevice/mtp/Makefile.am amarok-1.4.8_dst/amarok/src/mediadevice/mtp/Makefile.am
--- amarok-1.4.8_src/amarok/src/mediadevice/mtp/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/mediadevice/mtp/Makefile.am	2008-02-27 03:54:34.000000000 +0200
@@ -16,7 +16,7 @@
 
 libamarok_mtp_mediadevice_la_LIBADD  = \
     $(top_builddir)/amarok/src/libamarok.la \
-    $(LIBMTP_LIBS) 
+    $(LIBMTP_LIBS) $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI)
 
 libamarok_mtp_mediadevice_la_LDFLAGS = \
     $(KDE_PLUGIN) \
diff -uNr amarok-1.4.8_src/amarok/src/mediadevice/njb/Makefile.am amarok-1.4.8_dst/amarok/src/mediadevice/njb/Makefile.am
--- amarok-1.4.8_src/amarok/src/mediadevice/njb/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/mediadevice/njb/Makefile.am	2008-02-27 03:50:52.000000000 +0200
@@ -16,7 +16,7 @@
 kde_module_LTLIBRARIES = libamarok_njb-mediadevice.la
 
 libamarok_njb_mediadevice_la_LIBADD = $(top_builddir)/amarok/src/libamarok.la \
-	-lkdeui -lkdecore -lnjb
+	$(LIB_KDEUI) $(LIB_KDECORE) -lnjb $(LIB_QT) $(LIB_KIO)
 
 libamarok_njb_mediadevice_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries)
 
diff -uNr amarok-1.4.8_src/amarok/src/mediadevice/riokarma/Makefile.am amarok-1.4.8_dst/amarok/src/mediadevice/riokarma/Makefile.am
--- amarok-1.4.8_src/amarok/src/mediadevice/riokarma/Makefile.am	2008-02-27 03:47:00.000000000 +0200
+++ amarok-1.4.8_dst/amarok/src/mediadevice/riokarma/Makefile.am	2008-02-27 03:56:15.000000000 +0200
@@ -16,7 +16,7 @@
 
 libamarok_riokarma_mediadevice_la_LIBADD  = \
     $(top_builddir)/amarok/src/libamarok.la \
-    -ltag_c -lz -lusb -lkarma
+    -lkarma $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI)
 
 libamarok_riokarma_mediadevice_la_LDFLAGS = \
     $(KDE_PLUGIN) \
--- amarok-1.4.8_dst/amarok/configure.in.in	2007-12-17 17:53:19.000000000 +0200
+++ amarok-1.4.8_dst/amarok/configure.in.in	2008-02-27 11:15:25.000000000 +0200
@@ -573,7 +573,10 @@
 
     if test x$PKGCONFIGFOUND = xyes -a x$SDL_CONFIG = xyes; then
 
-        PKG_CHECK_MODULES(LIBVISUAL, libvisual-0.4 >= 0.4.0, [build_libvisual="yes"], [build_libvisual="no"])
+        PKG_CHECK_MODULES(LIBVISUAL, libvisual-0.4 >= 0.4.0, [
+                   LIBVISUAL_LIBS=`echo $LIBVISUAL_LIBS | sed -e 's/-lpthread//' -e 's/-ldl//'`
+                   build_libvisual="yes"
+        ], [build_libvisual="no"])
 
         AC_SUBST(LIBVISUAL_LIBS)
         AC_SUBST(LIBVISUAL_CFLAGS)
@@ -770,6 +773,7 @@
     if test "$PKGCONFIGFOUND" = "yes" ; then
         PKG_CHECK_MODULES(LIBMTP, libmtp >= 0.1.1,
         [
+            LIBMTP_LIBS=`echo ${LIBMTP_LIBS} | sed 's/-lusb//'`
             have_libmtp=yes
         ],
         [
