Index: amarok-1.4.10/debian/control
===================================================================
--- amarok-1.4.10.orig/debian/control	2009-06-08 00:30:55.000000000 +0200
+++ amarok-1.4.10/debian/control	2009-06-10 17:45:32.000000000 +0200
@@ -23,6 +23,7 @@
 Suggests: amarok14-engines, moodbar, konqueror | www-browser,
  python, python-qt3, libqt0-ruby1.8, libvisual-0.4-plugins
 Conflicts: amarok
+Provides: amarok
 Description: Amarok 1.4 series
  This is the last version of Amarok in the 1.4 series that was published
  in the Ubuntu repositories, before those switched to Amarok 2.*.
@@ -41,7 +42,8 @@
 Depends: ruby
 Recommends: amarok14 (>= ${source:Version})
 Suggests: libqt0-ruby1.8, python, python-qt3
-Conflicts: amarok14 (<= 2:1.4.9.1-0ubuntu3), amarok, amarok-common
+Conflicts: amarok14 (<= 2:1.4.9.1-0ubuntu3), amarok(>= 2:2), amarok-common
+Provides: amarok-common
 Description: architecture independent files for Amarok
  This package contains architecture independent files needed for Amarok to run
  properly. It also provides Amarok documentation. Therefore, unless you have
@@ -55,6 +57,7 @@
 Recommends: amarok14 (= ${binary:Version}), libxine1-ffmpeg
 Conflicts: amarok-xine, amarok-engine-xine
 Replaces: amarok-xine
+Provides: amarok-engine-xine
 Description: xine engine for Amarok 1.4 series
  This package provides the xine engine for Amarok 1.4.
  .
